feat: replace `micro-aes-gcm` with `@noble/chipers` [SPMVP-6208]

arielllin opened the pull request:
## Jira
https://storipress-media.atlassian.net/browse/SPMVP-6208

[//]: # 'This template is for new features, or changing an existing feature'
[//]: # 'Put the related jira links here'
[//]: # 'Please ensure there has description for the feature'

## Purpose
`micro-aes-gcm` 已經停止維護，更換為 `@noble/chipers`，並確認付費文章有正常加解密

[//]: # 'Please describe how the feature will affect user, e.g.'
[//]: # '- Allow to add link on image so publisher can directly link an image in static site'
[//]: # '- Adjust "Go to site" button to become "Go to Shopify" to let publisher using Shopify integration can directly go to see preview'
[//]: # (User should be one of "reader" {who read publisher's articles}, "publisher" {our users}, "developer" {us})

### For who

- [x] reader-facing?
- [ ] publisher-facing?
- [x] developer-facing?

## How did you make it?

[//]: # 'Give a brief for the changes you made'

## Production deployment notes

[//]: # 'Please describe the production deployment notes, e.g. this changes depend on other API or module change'

## 🎩 Tophat

Do a thorough 🎩. What is [tophatting](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md_)?

Consider testing:

- Existing functionality which could break due to your changes (e.g. global changes)
- New functionality being introduced. Ensure it meets intended behavior
- All different permutations (flows, states, conditions, etc) that are possible
- If you are modifying something which is used in multiple places, 🎩 all affected areas not just what you intend to change

### 🎩 Instructions

[//]: # "If it's complex, consider put a screen record here"

測試使用 `D6RX98VXN` 文章id `381` 這篇文章

1. 清除 local storage `storipress-token` 
2. 確認付費文章未顯示 ⭕ 
3. 增加 local storage `storipress-token` 模擬已登入狀態
4. 確認付費文章有顯示 ⭕ 

## Related PRs

[//]: # 'Put the related PRs here, e.g. PR in core-component'

## Checklist before requesting review

- [ ] I have done a self-review of my own code (comment on code is not required but recommended)
- [ ] I confirmed that I added review priority label (`priority:high`, `priority:medium`, ...)
- [ ] I did the Tophat steps and confirm the feature is working
- [ ] I have confirmed there is no issue reported by Static Analyzer (Please double check for custom class. For other issues, if you think it's ignorable, please add `// eslint-ignore-next-line <rule name>` on it)
- [ ] I have confirmed there is no deepsource issue (if you think it's ignorable, please explain why and add a [`skipcq` comment](https://deepsource.io/blog/releases-silence-issues-in-code/))
- [ ] I confirmed that the unit test is passed (if you break it, please fix it in this PR)

## Emoji Guide

<!-- from https://developers.soundcloud.com/blog/pr-templates-for-effective-pull-requests -->

**For reviewers: Emojis can be added to comments to call out blocking versus non-blocking feedback.**

E.g: Praise, minor suggestions, or clarifying questions that don’t block merging the PR.

> 🟢 Nice refactor!

> 🟡 Why was the default value removed?

E.g: Blocking feedback must be addressed before merging.

> 🔴 This change will break something important

|              |                |                                     |
| ------------ | -------------- | ----------------------------------- |
| Blocking     | 🔴 ❌ 🚨       | RED                                 |
| Non-blocking | 🟡 💡 🤔 💭    | Yellow, thinking, etc               |
| Praise       | 🟢 💚 😍 👍 🙌 | Green, hearts, positive emojis, etc |

## Gif (optional)

![image](https://i.gifer.com/origin/b8/b842107e63c67d5674d17e0f576274fa.gif)

arielllin committed:
feat: replace `micro-aes-gcm` with `@noble/chipers` [SPMVP-6208]

arielllin committed:
fix get uri undefined

arielllin commented on packages/karbon/src/runtime/composables/storipress-base-client.ts:
```diff
@@ -64,7 +64,7 @@ export function getStoripressConfig(): ModuleRuntimeConfig['storipress'] {
     const { storipress } = useRuntimeConfig()
     // @ts-expect-error type error
-    return storipress
+    return storipress ?? storipressConfigCtx.use()
```
這邊是修正取 storipress config 會 undefined 的問題

<img width="1412" alt="截圖 2024-02-01 上午10 42 11" src="https://github.com/storipress/karbon/assets/35054329/eab64b14-6322-4fe3-9d04-5dffa78da42a">

DanSnow commented on packages/karbon/src/runtime/api/normalize-article.ts:
```diff
@@ -73,6 +73,7 @@ export interface RawArticleLike {
 export interface PaidContent {
   content: string
   key: string
+  iv: number[]
```
這邊可以同樣做 base64 encode 與 decode 嗎？

arielllin replied:
調整好了

DanSnow commented on packages/karbon/src/runtime/api/article.ts:
```diff
@@ -347,17 +348,22 @@ async function encryptArticle({ plan, html, id, ...rest }: _NormalizeArticle) {
     paidContent = {
       key: encryptedKey,
       content: Buffer.from(content).toString('base64'),
+      iv: Array.from(iv),
```
可以參考 content 的方式做 base64 encode

arielllin replied:
調整好了

arielllin committed:
base64 encoded iv

deepsource-autofix[bot] committed:
style: format code with Prettier

This commit fixes the style issues introduced in 3f94918 according to the output
from Prettier.

Details: https://github.com/storipress/karbon/pull/324
